frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/VolumeClaimTemplateForm.tsx

```diff
@@ -0,0 +1,184 @@
+    const provisioner: string = updatedStorageClass?.provisioner || '';
+    // if the provisioner is unknown or no storage class selected, user should be able to set any access mode
+    const modes = provisionerAccessModeMapping[provisioner]
+      ? provisionerAccessModeMapping[provisioner]
+      : getAccessModeForProvisioner(provisioner);
+    // setting message to display for various modes when a storage class of a know provisioner is selected
+    const displayMessage =
+      provisionerAccessModeMapping[provisioner] || isCephProvisioner(provisioner)
+        ? t('pipelines-plugin~Access mode is set by storage class and cannot be changed')
+        : t('pipelines-plugin~Permissions to the mounted drive');
+    setAccessMode('ReadWriteOnce');
+    setAccessModeHelp(displayMessage);
+    // setting accessMode to default with the change to Storage Class selection
+    setAllowedAccessModes(modes);
+    setStorageClass(updatedStorageClass?.metadata?.name);
+    setStorageProvisioner(provisioner);
+    if (storageProvisioner.includes(cephStorageProvisioners[1])) {
+      setVolumeMode('Filesystem');
+    }
+  };
+
+  React.useEffect(() => {
+    const volumeClaimTemplate = {
+      spec: {
+        accessModes: [accessMode],
+        storageClassName: storageClass,
+        volumeMode,
+        resources: {
+          requests: {
+            storage: `${requestSizeValue}${requestSizeUnit}`,
+          },
+        },
+      },
+    };
+    setFieldTouched(name);
+    setFieldValue(name, volumeClaimTemplate);
+  }, [
+    volumeMode,
+    accessMode,
+    requestSizeValue,
+    requestSizeUnit,
+    storageClass,
+    name,
+    setFieldTouched,
+    setFieldValue,
+  ]);
+
+  return (
+    <>
+      <p className="help-block">
+        {t(
+          'pipelines-plugin~This will create a PersistentVolumeClaim with a size of {{requestSizeValue}} {{requestSizeUnit}}',
+          { requestSizeValue, requestSizeUnit },
+        )}
+      </p>
+      <ExpandCollapse
+        textExpanded={t('pipelines-plugin~Hide VolumeClaimTemplate options')}
+        textCollapsed={t('pipelines-plugin~Show VolumeClaimTemplate options')}
+      >
+        <StorageClassDropdown
+          onChange={handleStorageClass}
+          id="storageclass-dropdown"
+          data-test="storageclass-dropdown"
+          describedBy="storageclass-dropdown-help"
+          required={false}
+          valueFromAST
+          name="storageClass"
+        />
+        <label className="control-label co-required" htmlFor="access-mode">
+          {t('pipelines-plugin~Access Mode')}
+        </label>
+        <FormGroup fieldId="accessMode" data-test-id="accessModeRadio">
+          {accessModeRadios.map((radio) => {
+            const disabled = !allowedAccessModes.includes(radio.value);
+            let checked = null;
+            allowedAccessModes.forEach((mode) => {
+              checked = !disabled ? radio.value === accessMode : radio.value === mode;
+            });
```

Comment:
in the code piece 
` allowedAccessModes.forEach((mode) => {
     checked = !disabled ? radio.value === accessMode : radio.value === mode;
});`
wouldn't the expression `radio.value === mode` always yield false when disabled is true  i.e `allowedAccessModes` does not include radio.value?

Reply:
could we instead write it as
`const checked = disabled ? disabled : radio === accessMode;`

Reply:
What you say makes sense. But I've only replicated the logic used in create-pvc form without changing what they are trying to achieve.

I think in the case when all radios are disabled and one is selected by default. i.e user has no authority to make a change, this expression will hold true

Reply:
updated. Works fine locally

Reply:
I'm a bit confused how this will account for a default selection when every `radio` is disabled.
As far as i could understand, when `disabled` is true, every iteration of `forEach` evaluates the expression `radio.value === mode` which would never yield true since `disabled` is true only when `allowedAccessModes` does not include `radio.value`.